packages/cli/scripts/pre-designed-nft.ts

```diff
@@ -0,0 +1,244 @@
+  })
+
+  return config as NFTMetadataConfig
+}
+
+export async function validateTokenBaseUri(tokenBaseUri: string, maxSupply: number): Promise<NFTMetadata[]> {
+  return await validateTokenBaseUriForPreDesignedCollection(tokenBaseUri, maxSupply)
+}
+
+function createIPFSClient(projectId: string, projectSecret: string) {
+  const auth = 'Basic ' + Buffer.from(projectId + ':' + projectSecret).toString('base64')
+
+  return ipfsHttpClient({
+    host: 'ipfs.infura.io',
+    port: 5001,
+    protocol: 'https',
+    headers: {
+      authorization: auth
+    }
+  })
+}
+
+function validateStringTypeIfExists(parent: string, obj: object, key: string) {
+  if (!!obj[key] && typeof obj[key] !== 'string') {
+    throw new Error(`Key '${key}' under '${parent}' should be a string`)
+  }
+}
+
+function validateAttributesTypeIfExists(parent: string, attributes?: object) {
+  if (!!attributes) {
+    if (!Array.isArray(attributes)) {
+      throw new Error(`Field 'attributes' under '${parent}' should be an array`)
+    }
+
+    attributes.forEach((item) => {
+      if (typeof item !== 'object') {
+        throw new Error(`Field 'attributes' under '${parent}' should be an array of objects`)
+      }
+
+      Object.keys(item).forEach((key) => {
+        if (!validNFTMetadataAttributeTypes.includes(typeof item[key])) {
+          throw new Error(
+            `Field '${key}' in 'attributes' should be a string, boolean or number, but is ${typeof item[key]}`
+          )
+        }
+      })
+    })
+  }
+}
+
+async function uploadImageMetadataToIPFS(
+  ipfsDir: string,
+  metadataz: any[],
+  infuraProjectId: string,
+  infuraProjectSecret: string
+): Promise<string> {
+  const toBeUploaded = []
+  metadataz.forEach((metadata, index) => {
+    const remoteFilePath = path.join(ipfsDir, index.toString())
+    toBeUploaded.push({ path: remoteFilePath, content: JSON.stringify(metadata) })
+  })
+
+  if (toBeUploaded.length > 0) {
+    const ipfsClient = createIPFSClient(infuraProjectId, infuraProjectSecret)
+    let remoteDirURL: string | undefined = undefined
+    for await (const result of ipfsClient.addAll(toBeUploaded)) {
+      if (result.path === ipfsDir) {
+        remoteDirURL = `https://ipfs.io/ipfs/${result.cid.toString()}/`
+      }
+    }
+
+    return remoteDirURL
+  } else {
+    throw new Error(`Empty metadata`)
+  }
+}
+
+function parseNFTMetadataConfig(file: string, localImageDir: string): NFTMetadataConfig {
+  const content = fs.readFileSync(file, 'utf8')
+  const parsedContent = parse(content)
+  return validateMetadataConfig(parsedContent, localImageDir)
+}
+
+function convertAttributes(attributes: NFTMetadataConfig['attributes']): NFTMetadata['attributes'] | undefined {
+  if (attributes) {
+    return attributes.map((attribute) => {
+      const traitType = Object.keys(attribute)[0]
+      const value = attribute[traitType]
+      return { trait_type: traitType, value }
+    })
+  }
+}
+
+function padding(num: number, size: number): string {
```

Comment:
Nit: We can use the `padStart` function: `return num.toString().padStart(size, '0')`

Reply:
Sounds good :+1: